hera_cal/tests/test_frf.py

```diff
@@ -580,6 +580,50 @@ def test_load_tophat_frfilter_and_write(self, tmpdir):
                     assert not np.all(np.isclose(d[bl], 0.))
             np.testing.assert_array_equal(f[(53, 54, 'ee')], True)
             os.remove(outfilename)
+
+        # test wgt_by_nsample
+        uvh5 = os.path.join(DATA_PATH, "test_input/zen.2458101.46106.xx.HH.OCR_53x_54x_only.uvh5")
+        outfilename = os.path.join(tmp_path, 'temp.h5')
+        frf.load_tophat_frfilter_and_write(uvh5, res_outfilename=outfilename, tol=1e-4, clobber=True,
+                                           Nbls_per_load=None, avg_red_bllens=False, case='sky', wgt_by_nsample=True)
+        hd = io.HERAData(outfilename)
+        d, f, n = hd.read(bls=[(53, 54, 'ee')])
+        for bl in d:
+            assert not np.all(np.isclose(d[bl], 0.))
+
+        frfil = frf.FRFilter(uvh5, filetype='uvh5')
+        frfil.read(bls=[(53, 54, 'ee')])
+
+        wgts = datacontainer.DataContainer({k: (~frfil.flags[k]).astype(float) for k in frfil.flags})
+        for k in wgts:
+            wgts[k] *= frfil.nsamples[k]
+        frate_centers, frate_half_widths = frf.select_tophat_frates(case='sky', uvd=frfil.hd, blvecs=frfil.blvecs)
+        frfil.tophat_frfilter(keys=[(53, 54, 'ee')], tol=1e-4, verbose=True, frate_centers=frate_centers, frate_half_widths=frate_half_widths, wgts=wgts)
+        np.testing.assert_almost_equal(d[(53, 54, 'ee')], frfil.clean_resid[(53, 54, 'ee')], decimal=5)
+        np.testing.assert_array_equal(f[(53, 54, 'ee')], frfil.flags[(53, 54, 'ee')])
+
+        # test excluded_lsts
+        uvh5 = os.path.join(DATA_PATH, "test_input/zen.2458101.46106.xx.HH.OCR_53x_54x_only.uvh5")
+        outfilename = os.path.join(tmp_path, 'temp.h5')
+        frf.load_tophat_frfilter_and_write(uvh5, res_outfilename=outfilename, tol=1e-4, clobber=True,
+                                           Nbls_per_load=None, avg_red_bllens=False, case='sky', excluded_lsts=[(23, 6.03), (6.18, 7)])  # first and last five bins
+        hd = io.HERAData(outfilename)
+        d, f, n = hd.read(bls=[(53, 54, 'ee')])
+        for bl in d:
+            assert not np.all(np.isclose(d[bl], 0.))
+
+        frfil = frf.FRFilter(uvh5, filetype='uvh5')
+        frfil.read(bls=[(53, 54, 'ee')])
+
+        wgts = datacontainer.DataContainer({k: (~frfil.flags[k]).astype(float) for k in frfil.flags})
+        for k in wgts:
+            wgts[k][0:5, :] = 0
+            wgts[k][-5:, :] = 0
+        frate_centers, frate_half_widths = frf.select_tophat_frates(case='sky', uvd=frfil.hd, blvecs=frfil.blvecs)
+        frfil.tophat_frfilter(keys=[(53, 54, 'ee')], tol=1e-4, verbose=True, frate_centers=frate_centers, frate_half_widths=frate_half_widths, wgts=wgts)
+        np.testing.assert_almost_equal(d[(53, 54, 'ee')], frfil.clean_resid[(53, 54, 'ee')], decimal=5)
+        np.testing.assert_array_equal(f[(53, 54, 'ee')], frfil.flags[(53, 54, 'ee')])
+
```

Comment:
This looks like it should be its own test. I'm not sure what the testing philosophy is for this package, so feel free to ignore this comment, but I think it's best to make tests as granular as possible.

Reply:
This test already has a gazillion tests in it, so I'll just leave it for some future reorg.